### PR TITLE
Docker and package upgrades

### PR DESCRIPTION
**Pull request by mschwartz**
This PR contains:

1) npm upgrade for all modules except node sass which requires to be pinned to a 4.x.x version to work on OS X (at least OS X)
2) Dockerfile so the dev process can be done within a Docker container
3) docker-build.sh, docker-debug.sh, docker-prod-build.sh, and docker-clean.sh shell scripts for building, starting, and making production files.
4) Updated README.md to reflect option to use Docker.

Tested on Mac OS, was able to build production files and run/edit/debug files locally on the host while using the container.

NOTE: nothing is done for DevOps in any way.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "@ionic/react": "0.0.9",
     "classnames": "2.2.6",
     "lodash-es": "4.17.15",
+    "node-sass": "^4.14.1",
```
Comment (grgur) [inline package.json:20]: can we move this to `devDependencies`?

File: README.md [hunk 2]
```diff
@@ -23,6 +23,54 @@
...
+The benefit of Docker is that the specific versions of node_modules and Node JS required to build/run this 
+can be encapsulated within a Docker container rather than having to have those versions of software installed
+on your workstation.
+
+*NOTES* 
+1) Node version 12 seems to be required for node-sass to work.
+
+### Shell scripts
+
+The Dockerfile is used to build a Docker image that has no node_mocules/ directory.
```
Comment (Mrkezii) [inline README.md:37]: little typo here, i think it should be `node_modules`

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FROM node:12
```
Comment (Mrkezii) [inline Dockerfile:1]: if we are using node:12 here then i think we have to change the version of node in the [CI/CD config file](https://github.com/moduslabs/modir/blob/master/.circleci/config.yml#L30)

Comment (mschwartz) [reply Dockerfile:1]: I tried node:14, but the docker-prod-build.sh script failed due to node sass incompatibilities.  I tried uninstalling and reinstalling node sass to no avail.

If it works in Ci/CD config file as is, then I'm not sure why.  I am using MacOS (but Docker == Linux)

File: package.json (hunk 1 above)
Comment (mschwartz) [reply package.json]: Moved.

**Review by grgur — APPROVED** [review]

File: Dockerfile (hunk 3 above)
Comment (mschwartz) [reply Dockerfile:1]: https://stackoverflow.com/questions/37415134/error-node-sass-does-not-yet-support-your-current-environment-windows-64-bit-w

That kind of error with node > 12

Comment (Mrkezii) [reply Dockerfile:1]: oh, oh nice.

**Review by Mrkezii — APPROVED** [review]
Looks Good To Me

**Pull request merged**